### PR TITLE
expression: implement vectorized evaluation for `builtinLastInsertIDWithIDSig`

### PR DESCRIPTION
**Pull request by tangwz**
### What problem does this PR solve? 
Implement vectorized evaluation for builtinLastInsertIDWithIDSig.
Issue: #12103

### What is changed and how it works?
```
BenchmarkVectorizedBuiltinInfoFunc/builtinLastInsertIDWithIDSig-VecBuiltinFunc-4                  285380              4975 ns/op               0 B/op            0 allocs/op
BenchmarkVectorizedBuiltinInfoFunc/builtinLastInsertIDWithIDSig-NonVecBuiltinFunc-4                49749             22256 ns/op               0 B/op            0 allocs/op
```

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: expression/builtin_info_vec.go [hunk 1]
```diff
@@ -107,11 +107,21 @@ func (b *builtinLastInsertIDSig) vecEvalInt(input *chunk.Chunk, result *chunk.Co
...
-	return errors.Errorf("not implemented")
+	if err := b.args[0].VecEvalInt(b.ctx, input, result); err != nil {
+		return err
+	}
+	i64s := result.Int64s()
+	for i := 0; i < len(i64s); i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		b.ctx.GetSessionVars().SetLastInsertID(uint64(i64s[i]))
```
Comment (SunRunAway) [inline expression/builtin_info_vec.go]: Could we find the last not null `i` and just set once?

**Comment by Reminiscent** [comment]
@tangwz Thanks for your contribution. Please solve the conflict, thanks!

File: expression/builtin_info_vec.go (hunk 1 above)
Comment (ngaut) [reply expression/builtin_info_vec.go]: I am confused too.

Comment (qw4990) [reply expression/builtin_info_vec.go]: How about @tangwz : 
```
for i := len(i64s)-1; i >= 0; i-- {
    if !result.IsNull(i) {
        SetLastInsertID(...)
        break
    }
}
```

**Comment by tangwz** [comment]
@Reminiscent PTAL

**Pull request merged**